### DISCUSSION TITLE
Error while using Embedding Layer

### DISCUSSION THREAD

**Discussion by amank90** [truera/trulens#42] [Q&A]
Hi Team,

When there is an embedding layer in my model, i get attribution error "'NoneType' object has no attribute 'numpy'". I am not able to understand why it is giving me the error. I am using TF 2.5. Please help.

**Comment by rshih32** [comment]
Hi, 
This issue can happen for a variety of reasons, but usually stems from gradients returning None on the layer. Try running tf.gradients and see what it returns. We have found some tensorflow issues where gradients do not return for some layers, but that is related to the tensorflow library.

**Reply by amank90** [reply]
Thank you @rshih32 for the reply. I tried using torch and passed embedding layer on my categorical features along with my continuous features. I got the output from InputAttribution but attribution value for the input of my embedding layer is coming as zero. I also got non zero values for my continuous input features which seams fine. I am not sure why it is returning zero value for the input of embedding layer.

**Reply by rshih32** [reply]
One thought is that the embedding layer is not the input layer of your model, and is an intermediate layer. Try using the InternalAttribution class and set the first cut to the 'in' of your embedding layer. If that still didn't work, it might be useful for us to see some code to repro the issue
